sys/Makefile.dep

```diff
@@ -259,6 +263,10 @@ ifneq (,$(filter posix_sockets,$(USEMODULE)))
   USEMODULE += posix_headers
 endif
 
+ifneq (,$(filter printf_float,$(USEMODULE)))
+  FEATURES_REQUIRED += float_math
+endif
```

Comment:
That is actually done here. But most apps that do print float/double also do math with float/double before printing it.

It would be possible to actually implement the `%f` format specifier in `printf` without using floating point arithmetic directly. But then those apps would still not work on systems that do not provide floating point arithmetic if the use float math themselves.